haskell/cabal.bzl

```diff
@@ -2561,6 +2563,12 @@ def stack_snapshot(
 
     """
     typecheck_stackage_extradeps(extra_deps)
+
+    # Allow overriding stack binary by external repositories.
+    # Otherwise this is a no-op.
+    if existing_rule("rules_haskell_stack"):
```

Comment:
IIUC this would also override a user provided `stack` attribute. Checking `if not stack and existing_rule(...)` should fix that.

Reply:
I thought this was the point of #1080 - allowing an override?

Reply:
Good catch, I overlooked
> if stack_snapshot provides a stack argument, no way to change it

Let's keep it this way then.